src/adagrams.js

```diff
@@ -1,15 +1,142 @@
+//WAVE 1
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  const LETTER_POOL = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+
+  let letterBank = [];
+  let letterCopy = JSON.parse(JSON.stringify(LETTER_POOL));
+  let keyList = Object.keys(letterCopy);
 
+  while (letterBank.length < 10) {
+    let randomLetter = keyList[Math.floor(Math.random() * keyList.length)];
+    if (letterCopy[randomLetter] > 0) {
+      letterBank.push(randomLetter);
+      letterCopy[randomLetter]--;
+    }
```

Comment:
This works and is roughly 0(n^2) in terms of time/space complexity, however, you can do this in a more efficient manner by using a different version of looping such as a for loop so that you can avoid having to do a check for if the random copy is greater than zero:
```
  // Create a loop to draw 10 letters from the letter pool
  for (let i = 0; i < 10; i++) {
    // Grab a random index from the letter pool
    let n = letterPoolCopy.length;
    let i = Math.floor(n * Math.random());
```